src/query/vacuum.c

```diff
@@ -4085,13 +4087,14 @@ vacuum_data_load_and_recover (THREAD_ENTRY * thread_p)
   vacuum_Data.last_page = data_page;
   data_page = NULL;
   /* Get last_blockid. */
-  if (vacuum_Data.last_page->index_unvacuumed == vacuum_Data.last_page->index_free)
+  if (vacuum_is_empty ())
     {
-      /* Empty last page. */
-      assert (vacuum_Data.last_page->index_unvacuumed == 0);
-      /* last_blockid is still saved in the first data entry. */
-      vacuum_Data.last_blockid = vacuum_Data.last_page->data->blockid;
-      assert (vacuum_Data.last_blockid == VACUUM_BLOCKID_WITHOUT_FLAGS (vacuum_Data.last_blockid));
+      /* Get the maximum between what is currently stored in vacuum and the value stored
+       * in the log_Gl header. After a long session in SA_MODE, the vacuum_Data.last_page->data->blockid will not
```

Comment:
Rephrase as "After a long session in SA_MODE, the vacuum_Data.last_page->data->blockid will be outdated. Instead SA_MODE updates log_Gl.hdr.vacuum_last_blockid before removing old archives.".